Jasmine/Common/Models/CharacterGrid.swift

```diff
@@ -0,0 +1,83 @@
+struct CharacterGrid {
+    /// The actual grid.
+    private var grid: [[String?]]
```

Comment:
So that there are bounds to the data structure. With 3 rows and 3 cols, `grid[7]` will crash while with the dictionary, `grid[Coordinate(row: 7, col: 7)]` will return `nil`, just like `grid[Coordinate(row: 0, col: 0)]` can also return nil.